File: NetcraftServer/Commands/Commandkill.vb
Imports NetcraftServer

Public Class Commandkill
    Inherits Command
    Sub New()
        MyBase.New("kill", "Убить указанного игрока", "kill <игрок>")
    End Sub
    Public Overrides Function OnCommand(sender As CommandSender, cmd As Command, args() As String, label As String) As Boolean
        If Not sender.GetAdmin Then
            sender.SendMessage("У Вас недостаточно прав!")
            Return True
        End If
        If args.Length = 1 Then
            Dim p As NetworkPlayer
            p = netcraft.server.api.NCSApi.Netcraft.GetPlayer(args(0))
            If IsNothing(p) Then
                sender.SendMessage("Игрок не найден!")
                Return True
            End If
            p.Kill("was slain!")
            sender.SendMessage("Slain: " + args(0))
            Return True
        End If
        Return False
    End Function
End Class

File: NetcraftCore/Commands/Commandkill.cs
using NCore;
using NCore.netcraft.server.api;
using System.Threading.Tasks;

namespace NCore
{
    public class Commandkill : Command
    {
        public Commandkill() : base("kill", NCore.GetNCore().lang.get("commands.kill.description"), "netcraft.command.kill", NCore.GetNCore().lang.get("commands.kill.usage"))
        {
        }

        public override async Task<bool> OnCommand(CommandSender sender, Command cmd, string[] args, string label)
        {
            NCore.Lang lang = sender.IsPlayer ? ((NetcraftPlayer)sender).lang : NCore.GetNCore().lang;

            if (args.Length == 1)
            {
                NetcraftPlayer p;
                p = Netcraft.GetPlayer(args[0]);
                if (NCore.IsNothing(p))
                {
                    await sender.SendMessage(lang.get("commands.generic.player.not-found"));
                    return true;
                }
                await p.UpdateHealth(0, "was slain by admin");
                await NCore.GetNCore().SendCommandFeedback(args[0] + " killed", sender);
                return true;
            }

            return false;
        }
    }
}